### PR TITLE
Throw friendly error when device is disconnected

### PR DESCRIPTION
**Pull request by WasabiFan**

### CODE REVIEW

File: ev3dev/core.py [hunk 1]
```diff
@@ -225,6 +225,8 @@ def _raise_friendly_access_error(self, driver_error, attribute):
                 else:
                     raise ValueError("The given speed value was out of range. Max speed: +/-" + str(max_speed)) from driver_error
             raise ValueError("One or more arguments were out of range or invalid") from driver_error
+        elif driver_error.errno == errno.ENODEV:
+            raise Exception("The device is no longer connected") from driver_error
```
Comment (ddemidov) [inline ev3dev/core.py]: Why 'no longer'? Do we know it was ever connected?

Comment (WasabiFan) [reply ev3dev/core.py]: Currently, if it was never connected, the `connected` property is false and we throw an exception at a higher level. There's an `if` somewhere around  here which handles that case.

Comment (dwalton76) [reply ev3dev/core.py]: How about:
```
raise Exception("%s is no longer connected" % self) from driver_error
```
that way the exception has the `__str__` of the device that was disconnected

File: ev3dev/core.py [hunk 2]
```diff
@@ -197,12 +197,12 @@ def _get_attribute(self, attribute, name):
     def _set_attribute(self, attribute, name, value):
         """Device attribute setter"""
         if self.connected:
-            if None == attribute:
-                attribute = self._attribute_file_open( name )
-            else:
-                attribute.seek(0)
-
             try:
+                if None == attribute:
```
Comment (dwalton76) [inline ev3dev/core.py]: Do we follow PEP-8?  If so we should do `if attribute is None` here.

File: ev3dev/core.py (hunk 1 above)
Comment (WasabiFan) [reply ev3dev/core.py]: I haven't paid attention to how inheritance works in Python... Will that have the expected behavior? I know that `__class__` is the type the method was defined in and not the runtime type of the enclosing object.

File: ev3dev/core.py (hunk 2 above)
Comment (WasabiFan) [reply ev3dev/core.py]: Yeah, this is existing code. I'll update it.

File: ev3dev/core.py (hunk 1 above)
Comment (dwalton76) [reply ev3dev/core.py]: The Device class has a `__str__` that will be used to determine what to print whenver you have to print the object in a string

Comment (WasabiFan) [reply ev3dev/core.py]: The question I was asking was a technical one about the language. I'll figure it out and update as needed.

**Review by dwalton76 — APPROVED** [review]
Looks good

**Pull request merged**